### PR TITLE
Fixes part of #40 & #42: Button extreme margins

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes part of #40
Fixes part of #42

This PR introduces the dimens value for start/end margins for buttons which will be used in later PRs.

Also, check below comments where I have mentioned reference for each value.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -399,4 +399,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">44dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">44dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Exploration View -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:407]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/3c9aff8d-2319-4570-818e-cbef229a7419/specs/

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -399,4 +399,20 @@
...
+  <dimen name="general_button_item_exploration_view_margin_start">28dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">28dp</dimen>
+
+  <!-- General Button: Question Split View -->
+  <dimen name="general_button_item_question_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_question_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Question View -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:415]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/b6d8fedf-85aa-4052-9b7a-5d6be7f03e01/specs/

File: app/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -399,4 +399,20 @@
...
+
+  <!-- General Button Item: Exploration Split View -->
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Exploration View -->
+  <dimen name="general_button_item_exploration_view_margin_start">28dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">28dp</dimen>
+
+  <!-- General Button: Question Split View -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:411]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/f97ba4a6-b92f-4388-b1bd-0a80f49d552d/specs/

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -399,4 +399,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">44dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">44dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:403]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/7d089aae-3288-4474-8e4b-14a6f67878f3/specs/

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 5]
```diff
@@ -200,4 +200,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">140dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">140dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Exploration View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml:208]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/986ffd25-9e86-42a2-b870-aa6e754799e4/specs/

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 6]
```diff
@@ -200,4 +200,20 @@
...
+  <dimen name="general_button_item_exploration_view_margin_start">112dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">112dp</dimen>
+
+  <!-- General Button: Question Split View -->
+  <dimen name="general_button_item_question_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_question_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Question View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml:216]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/243dda5c-50b8-4666-b3f2-22bbd7e6023a/specs/

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 7]
```diff
@@ -200,4 +200,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">140dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">140dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml:204]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/7d089aae-3288-4474-8e4b-14a6f67878f3/specs/

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 8]
```diff
@@ -200,4 +200,20 @@
...
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">48dp</dimen>
+
+  <!-- General Button: Exploration View -->
+  <dimen name="general_button_item_exploration_view_margin_start">112dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">112dp</dimen>
+
+  <!-- General Button: Question Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml:212]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/f97ba4a6-b92f-4388-b1bd-0a80f49d552d/specs/

File: app/src/main/res/values-land/dimens.xml [hunk 9]
```diff
@@ -149,4 +149,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">76dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">76dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:153]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/8b4be174-68ae-4b99-b937-9f517c6799d6/specs/

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 10]
```diff
@@ -197,4 +197,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">204dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">204dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Exploration View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml:205]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/9b11d6ac-ba50-4d28-b355-28916363cfc9/specs/

File: app/src/main/res/values-land/dimens.xml [hunk 11]
```diff
@@ -149,4 +149,20 @@
...
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Exploration View -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:157]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/8b4be174-68ae-4b99-b937-9f517c6799d6/specs/

File: app/src/main/res/values-land/dimens.xml [hunk 12]
```diff
@@ -149,4 +149,20 @@
...
+  <dimen name="general_button_item_exploration_view_margin_start">64dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Question Split View -->
+  <dimen name="general_button_item_question_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_question_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Question View -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:165]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/acb4086c-cdd8-4d4d-a909-a2d18e1472fc/specs/

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 13]
```diff
@@ -197,4 +197,20 @@
...
+  <dimen name="general_button_item_exploration_view_margin_start">176dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">176dp</dimen>
+
+  <!-- General Button: Question Split View -->
+  <dimen name="general_button_item_question_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_question_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Question View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml:213]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/01adc357-dec8-4f7a-b639-a9aafa89f3f3/specs/

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 14]
```diff
@@ -197,4 +197,20 @@
   <!-- Previous Responses Item: Question View -->
   <dimen name="previous_responses_item_question_view_margin_start">204dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_end">204dp</dimen>
+
+  <!-- General Button Item: Exploration Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml:201]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/60ac31c1-6ac5-4a05-b79c-8702a095cc39/specs/

File: app/src/main/res/values-land/dimens.xml (hunk 9 above)
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:153]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/60ac31c1-6ac5-4a05-b79c-8702a095cc39/specs/

File: app/src/main/res/values-land/dimens.xml [hunk 15]
```diff
@@ -149,4 +149,20 @@
...
+
+  <!-- General Button Item: Exploration Split View -->
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Exploration View -->
+  <dimen name="general_button_item_exploration_view_margin_start">64dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Question Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml:161]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/a32ccabf-4ae8-4112-a021-333f0d0466d6/specs/

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 16]
```diff
@@ -197,4 +197,20 @@
...
+  <dimen name="general_button_item_exploration_split_view_margin_start">48dp</dimen>
+  <dimen name="general_button_item_exploration_split_view_margin_end">64dp</dimen>
+
+  <!-- General Button: Exploration View -->
+  <dimen name="general_button_item_exploration_view_margin_start">176dp</dimen>
+  <dimen name="general_button_item_exploration_view_margin_end">176dp</dimen>
+
+  <!-- General Button: Question Split View -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml:209]: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/a32ccabf-4ae8-4112-a021-333f0d0466d6/specs/

File: app/src/main/res/values-land/dimens.xml (hunk 11 above)
Comment (anandwana001) [reply app/src/main/res/values-land/dimens.xml:157]: Correct

File: app/src/main/res/values-land/dimens.xml (hunk 9 above)
Comment (anandwana001) [reply app/src/main/res/values-land/dimens.xml:153]: Correct

File: app/src/main/res/values-land/dimens.xml (hunk 15 above)
Comment (anandwana001) [reply app/src/main/res/values-land/dimens.xml:161]: Correct

File: app/src/main/res/values-land/dimens.xml (hunk 12 above)
Comment (anandwana001) [reply app/src/main/res/values-land/dimens.xml:165]: Correct

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 14 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-land/dimens.xml:201]: Correct

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 10 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-land/dimens.xml:205]: Correct

Comment (anandwana001) [reply app/src/main/res/values-sw600dp-land/dimens.xml:205]: Looks a bit odd, the start margin is the same as the content, which makes them display in a similar margin of 176, but the end, the button has 176 and the content has 208. 

But, in another exploration view, we tend to keep the button more margin so it comes inside than content.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 16 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-land/dimens.xml:209]: correct

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 13 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-land/dimens.xml:213]: correct

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 7 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-port/dimens.xml:204]: correct

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 5 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-port/dimens.xml:208]: Do we need these values?

As this is a port res dir, so if they will always use the split view, the split resources will be used. I think we can remove these items!! Let me know if I miss anything here.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 8 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-port/dimens.xml:212]: correct

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 6 above)
Comment (anandwana001) [reply app/src/main/res/values-sw600dp-port/dimens.xml:216]: correct

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (anandwana001) [reply app/src/main/res/values/dimens.xml:403]: The same applies to here, this is the mobile res directory so we might not be using the split resources here.

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (anandwana001) [reply app/src/main/res/values/dimens.xml:407]: correct

File: app/src/main/res/values/dimens.xml (hunk 2 above)
Comment (anandwana001) [reply app/src/main/res/values/dimens.xml:415]: correct

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Nit thoughts left, Thanks @rt4914

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 10 above)
Comment (rt4914) [reply app/src/main/res/values-sw600dp-land/dimens.xml:205]: For mobile we are generally keeping the buttons a bit inward w.r.t. overall content but in tablet we are keeping them outward. The outward is more correct but for mobile we need to make sure that there is sufficient margins on all side and thats why we kept it inwards.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 5 above)
Comment (rt4914) [reply app/src/main/res/values-sw600dp-port/dimens.xml:208]: Actually Sliptview depends on two things: (a.) Physical size of more than 7" and (b.) DragAndDrop/ImageRegionSelection interaction

It is independent of tablet or mobile or its configuration

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/values/dimens.xml:403]: Actually Sliptview depends on two things: (a.) Physical size of more than 7" and (b.) DragAndDrop/ImageRegionSelection interaction

It is independent of tablet or mobile or its configuration

So there might come a mobile which is more than 7" and in that case will need to use split screen.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Pull request merged**